ext/opentelemetry-ext-opentracing-shim/tests/test_shim.py

```diff
@@ -0,0 +1,386 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import unittest
+
+import opentracing
+
+import opentelemetry.ext.opentracingshim as opentracingshim
+from opentelemetry import trace
+from opentelemetry.ext.opentracingshim import util
+from opentelemetry.sdk.trace import Tracer
+
+
+class TestShim(unittest.TestCase):
+    # pylint: disable=too-many-public-methods
+
+    def setUp(self):
+        # TODO: Review setup.
+        self.tracer = trace.tracer()
+        self.shim = opentracingshim.create_tracer(self.tracer)
+
+    @classmethod
+    def setUpClass(cls):
+        """Set preferred tracer implementation only once rather than before
+        every test method.
+        """
+        # TODO: Do we need to call setUpClass() on super()?
+        # Seems to work fine without it.
```

Comment:
I don't think so, they're [no-ops in `TestCase`](https://github.com/python/cpython/blob/fd628cf5adaeee73eab579393cdff71c8f70cdf2/Lib/unittest/case.py#L488-L494).